src/main/java/org/jabref/gui/ClipBoardManager.java

```diff
@@ -89,40 +88,55 @@ public void setContent(List<BibEntry> entries) throws IOException {
         clipboard.setContent(content);
     }
 
-    public List<BibEntry> extractEntries() {
+    public List<BibEntry> extractData() {
         Object entries = clipboard.getContent(DragAndDropDataFormats.ENTRIES);
 
+        if (entries == null) {
+            return handleStringData(clipboard.getString());
+        }
+        return handleBibTeXData((String) entries);
+    }
+
+    private List<BibEntry> handleBibTeXData(String entries) {
         BibtexParser parser = new BibtexParser(Globals.prefs.getImportFormatPreferences(), Globals.getFileUpdateMonitor());
-        if (entries != null) {
-            // We have determined that the clipboard data is a set of entries (serialized as a string).
-            try {
-                return parser.parseEntries((String) entries);
-            } catch (ParseException ex) {
-                LOGGER.error("Could not paste", ex);
-            }
-        } else {
-            String data = clipboard.getString();
-            if (data != null) {
-                try {
-                    // fetch from doi
-                    Optional<DOI> doi = DOI.parse(data);
-                    if (doi.isPresent()) {
-                        LOGGER.info("Found DOI in clipboard");
-                        Optional<BibEntry> entry = new DoiFetcher(Globals.prefs.getImportFormatPreferences()).performSearchById(doi.get().getDOI());
-                        return OptionalUtil.toList(entry);
-                    } else {
-                        try {
-                            UnknownFormatImport unknownFormatImport = importFormatReader.importUnknownFormat(data);
-                            return unknownFormatImport.parserResult.getDatabase().getEntries();
-                        } catch (ImportException e) {
-                            // import failed and result will be empty
-                        }
-                    }
-                } catch (FetcherException ex) {
-                    LOGGER.error("Error while fetching", ex);
-                }
-            }
+        try {
+            return parser.parseEntries(new ByteArrayInputStream(entries.getBytes(StandardCharsets.UTF_8)));
```

Comment:
Hi Stefan,
Nice to see some commits from you ;-)
I'm not sure whether this is not breaking then the parsing from Strings in other encodings... Have you Hi checked it with some ISO encoded special characters?

Reply:
ISO special characters should be mappable in UTF-8, right?! I guess currentyl the default encoding for copying to the clipboard is UTF-8, so pasting with UTF-8 made sense to me. If we want to preserve the real encoding, we probably need to pass the encoding with the DataFlavor somehow? Like https://stackoverflow.com/questions/29651301/text-copied-from-jtextarea-have-broken-encoding-after-paste-in-foxpro-applicatio

Reply:
We always use UTF-8 internally, so this makes indeed sense. Only use case: What if the users has selected a different encoding for the database in the prefs?

Reply:
> If we want to preserve the real encoding, we probably need to pass the encoding with the DataFlavor somehow? Like https://stackoverflow.com/questions/29651301/text-copied-from-jtextarea-have-broken-encoding-after-paste-in-foxpro-applicatio

I guess we need to do something like this then.

Reply:
This is old swing stuff, don't do this. 
It should be sufficient to use the encoding in this method and apply the same encoding when pasting. 
https://github.com/JabRef/jabref/blob/eb42850f7853aff1f98459e533bc254217a2fa8d/src/main/java/org/jabref/gui/ClipBoardManager.java#L83-L89

Reply:
> This is old swing stuff, don't do this.

I know. I'm just talking about passing encoding somehow. Currently, I don't see any option with the JavaFX API.

Reply:
I cannot spot any place during serialization where we use an encoding different to the default encoding which seems to be UTF-8 then.

Reply:
I don't we need to pay attention to the database encoding since this option only controls the encoding when the database is written to the file (and does not affect the entries in itself). 

It may be worth a try to use `parser.parseEntries(new StringReader(entries))` which would circumvent the need to specify an encoding. But whether this works depends on if the JavaFX clipboard is able to correctly encode/decode the string (I tried to dig into the javafx code but it seems that they let the os handle the encoding). Otherwise your solution looks good to me.